style-guide/check-same.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
```

Comment:
There are places where this word is perfectly fine. I wonder if there's a way to narrow it down to likely problems...

At any rate, it doesn't always crop up as `The same`.

Reply:
As of now, I cannot find a way to crop up as ``The same``. Maybe we could avoid ignoring case to crop to some cases or just remove this test? What do you suggest?

Reply:
I made some changes to the test. Now the check only matches ``The same`` after a period and space and it is case-sensitive as well.